packages/cdktf/lib/complex-computed-list.ts

```diff
@@ -17,11 +21,19 @@ abstract class ComplexResolvable implements IResolvable, ITerraformAddressable {
     this.creationStack = captureStackTrace();
   }
 
-  abstract fqn: string;
+  abstract computeFqn(): string;
+
+  get fqn(): string {
+    if (!this._fqn) {
+      this._fqn = this.computeFqn();
```

Comment:
Wouldn't this be a great place to use a lazy token instead?

Reply:
No. Lazy results in failed cross stack references due to how / when it is being resolved.